core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -570,6 +423,190 @@ class TiBatchWrite(@transient val df: DataFrame,
 
     TableCodec.encodeRow(tiTableInfo.getColumns, convertedValues, tiTableInfo.isPkHandle)
   }
+
+  private def generateRDDToBeInserted(
+    rdd: RDD[toBeCheckedRow]
+  ) = {
+    var allocatedRdd: RDD[(toBeCheckedRow, Long)] = null
+    if (!tiTableInfo.isPkHandle) {
+      val step = rdd.count
+      val catalog = TiSessionCache.getSession(tiConf).getCatalog
```

Comment:
add `private var catalog: Catalog = _` to line 84?

Reply:
Catalog can not be serialized.

Reply:
add `@transient private var catalog: Catalog = _ `